mindmeld/models/tagger_models.py

```diff
@@ -324,78 +357,91 @@ def evaluate(self, examples, labels):
-                CRF_TYPE: ConditionalRandomFields,
-                LSTM_TYPE: LstmModel,
-            }[classifier_type]
-        except KeyError as e:
-            msg = "{}: Classifier type {!r} not recognized"
-            raise ValueError(msg.format(self.__class__.__name__, classifier_type)) from e
+    def _dump(self, path):
 
-    def dump(self, path, config):
-        """
-        Dumps the model using joblib for the config and call's the underlying model
-        to dump its state.
+        # In TaggerModel, unlike TextModel, two dumps happen,
+        # one, the underneath classifier and two, the model metadata
 
-        Args:
-            path (str): The path to dump the model to
-            config (dict): The config containing the model configuration
-        """
-        self._clf.dump(path, config)
+        os.makedirs(os.path.dirname(path), exist_ok=True)
+        metadata = {"serializable": self._clf.is_serializable}
 
-        if "model" not in config:
-            # If the model path is not populated, the model is serializable, so
-            # we just pass the entire model to the dictionary
-            config["model"] = self
+        if self._clf.is_serializable:
+            metadata.update({
+                "model": self
+            })
         else:
-            variables_to_dump = {
+            # underneath tagger dump (eg. LSTM)
+            model_dir = self._clf.dump(path)
+            metadata.update({
+                "model": model_dir,
                 "current_params": self._current_params,
                 "label_encoder": self._label_encoder,
                 "no_entities": self._no_entities,
-            }
-            joblib.dump(
-                variables_to_dump, os.path.join(config["model"], ".tagger_vars")
-            )
-
-        joblib.dump(config, path)
+            })
 
-    def unload(self):
-        self._clf = None
-        self._current_params = None
-        self._label_encoder = None
-        self._no_entities = None
+        # dump model metadata
+        joblib.dump(metadata, path)
 
-    def load(self, path, config):
+    @classmethod
+    def load(cls, path):
         """
-        Load the model state to memory. This method is strictly for non-serializable models
-        like tensorflow models
+        Load the model state to memory.
 
         Args:
             path (str): The path to dump the model to
-            config (dict): The config containing the model configuration
         """
-        self._clf.load(path)
-        variables_to_load = joblib.load(os.path.join(config["model"], ".tagger_vars"))
-        self._current_params = variables_to_load["current_params"]
-        self._label_encoder = variables_to_load["label_encoder"]
-        self._no_entities = variables_to_load["no_entities"]
 
-    def get_feature_matrix(self, examples, y=None, fit=False):
-        raise NotImplementedError
+        # load model metadata
+        metadata = joblib.load(path)
```

Comment:
same for the load functionality consistency as well
```suggestion
        tagger_data = joblib.load(path)
```